biojava-ontology/src/test/java/org/biojava/nbio/ontology/TestOboFileParsing.java

```diff
@@ -37,23 +37,40 @@
 
 public class TestOboFileParsing {
 
-	private static final Logger logger = LoggerFactory.getLogger(TestOboFileParsing.class);
+    private static final Logger logger = LoggerFactory.getLogger(TestOboFileParsing.class);
 
-	@Test
-	public void testParsingBioSapiensOBO() throws Exception {
-		OboParser parser = new OboParser();
-		InputStream inStream = parser.getClass().getResourceAsStream("/ontology/biosapiens.obo");
+    @Test
+    public void testParsingBioSapiensOBO() throws Exception {
+        OboParser parser = new OboParser();
+        InputStream inStream = parser.getClass().getResourceAsStream("/ontology/biosapiens.obo");
 
-		Assert.assertNotNull(inStream);
+        Assert.assertNotNull(inStream);
 
-		BufferedReader oboFile = new BufferedReader ( new InputStreamReader ( inStream ) );
+        BufferedReader oboFile = new BufferedReader ( new InputStreamReader ( inStream ) );
 
-		Ontology ontology;
+        Ontology ontology;
 
-		ontology = parser.parseOBO(oboFile, "BioSapiens", "the BioSapiens ontology");
-		Set<Term> keys = ontology.getTerms();
+        ontology = parser.parseOBO(oboFile, "BioSapiens", "the BioSapiens ontology");
+        Set<Term> keys = ontology.getTerms();
 
-		Assert.assertTrue(keys.size() > 4000);
-	}
+        Assert.assertTrue(keys.size() > 4000);
+    }
+
+    @Test
+    public void testParsingHPOOBO() throws Exception {
+        OboParser parser = new OboParser();
+        InputStream inStream = parser.getClass().getResourceAsStream("/ontology/hp.obo");
```

Comment:
If you gzip the hp.obo file, here you just need to use the gzip filter to read it, i.e.
```java
InputStream inStream = new GZIPInputStream(parser.getClass().getResourceAsStream("/ontology/hp.obo"));
```